.github/workflows/workflow.yml

```diff
@@ -0,0 +1,58 @@
+name: OpenSSL
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    name: >-
+      OS: ${{ matrix.os }}  Ruby: ${{ matrix.ruby }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ 'ubuntu-16.04', 'ubuntu-18.04', 'macos', 'windows-latest' ]
+        ruby: [ '2.3.x', '2.4.x', '2.5.x', '2.6.x' ]
+        exclude:
+        - os: ubuntu-16.04
+          ruby: 2.4.x
+        - os: ubuntu-16.04
+          ruby: 2.5.x
+        - os: ubuntu-16.04
+          ruby: 2.6.x
+        - os: ubuntu-18.04
+          ruby: 2.3.x
+        - os: macos
+          ruby: 2.3.x
+        - os: windows-latest
+          ruby: 2.3.x
+    steps:
+    - name: repo checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 10
```

Comment:
Why 10 and not 1 as in [the example](https://help.github.com/en/github/automating-your-workflow-with-github-actions/configuring-a-workflow#using-the-checkout-action)? We're not going to be doing anything with the history?

Reply:
Because several CI builds could be queued up at the same time.

Reply:
According to https://github.com/actions/checkout and https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows every invocation of the action checks out a specific SHA, which includes any previous changes up to that point. I don't see how those other 9 commits will be used no matter any queueing going on?

Reply:
https://github.com/actions/checkout/blob/master/action.yml#L17-L18

Specifically, 'defaults to no limit'

Reply:
Sorry, earlier I was distracted.  Without fetch-depth, all commit history is retrieved via git fetch.  So, it should be larger than the maximum number of commits that one expects to be in the CI 'pipeline' at an given time.

Without it, large repos can take longer to retrieve...

Reply:
I'm not arguing against doing a shallow clone, but against using a depth of 10 instead of 1 🙂 Every queued build will run the action and _get their own history_ up to the SHA checked out. There's no need for every build fetching 9 more commits of history. They won't be used, there's no sharing going on. 

Also see https://github.com/actions/checkout/issues/22 where further optimizations are discussed.

Reply:
I think ruby/ruby did their own command, similar to what's shown, as they have a lot of tags & commits.

Regardless, the longest checkout step when run in my fork took 5 seconds.  We've spent more time here than the next several commits/pr's wound take...

Reply:
@bdewater please submit an updated PR after this merged and we can confirm your thoughts, thanks everyone for the discussion and work.